### PR TITLE
Add go_genrule

### PR DESCRIPTION
**Pull request by mikedanese**
Original PR here https://github.com/kubernetes/release/pull/238

### CODE REVIEW

**Comment by spxtr** [comment]
Why should this go here rather than in k/k?

**Comment by mikedanese** [comment]
@spxtr we will need it in other repositiories if we break stuff out of kubernetes. That's the purpose of this repo.

**Comment by spxtr** [comment]
I don't see anything offensive here.

LGTM

File: defs/go.bzl [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            cfg = "host",
+            allow_files = True,
+        ),
+        "outs": attr.output_list(mandatory = True),
+        "cmd": attr.string(mandatory = True),
+        "go_deps": attr.label_list(
+            aspects = [go_sources_aspect],
+        ),
+        "message": attr.string(),
+        "executable": attr.bool(default = False),
```
Comment (ixdy) [inline defs/go.bzl]: it looks like this isn't used? or is this somehow magically used elsewhere?

**Pull request merged**